packages/@uppy/dashboard/src/components/FileItem/index.js

```diff
@@ -1,76 +1,89 @@
-const { h } = require('preact')
+const { h, Component } = require('preact')
 const classNames = require('classnames')
-const pure = require('../../utils/pure')
+const shallowEqual = require('is-shallow-equal')
 const FilePreviewAndLink = require('./FilePreviewAndLink')
 const FileProgress = require('./FileProgress')
 const FileInfo = require('./FileInfo')
 const Buttons = require('./Buttons')
 
-module.exports = pure(function FileItem (props) {
-  const file = props.file
+module.exports = class FileItem extends Component {
+  shouldComponentUpdate (nextProps) {
+    return !shallowEqual(this.props, nextProps)
+  }
 
-  const isProcessing = file.progress.preprocess || file.progress.postprocess
-  const isUploaded = file.progress.uploadComplete && !isProcessing && !file.error
-  const uploadInProgressOrComplete = file.progress.uploadStarted || isProcessing
-  const uploadInProgress = (file.progress.uploadStarted && !file.progress.uploadComplete) || isProcessing
-  const isPaused = file.isPaused || false
-  const error = file.error || false
+  componentDidMount () {
+    const file = this.props.file
+    if (!file.preview && this.props.handleRequestThumbnail) {
```

Comment:
Why do we need `&& this.props.handleRequestThumbnail` conditional, isn't it always defined?

Reply:
good spot, i was initially going to only pass it in if lazy generation was enabled, so this is a leftover